http/write_handler.go

```diff
@@ -210,22 +238,28 @@ func (h *WriteHandler) handleWrite(w http.ResponseWriter, r *http.Request) {
 	span.Finish()
 	if err != nil {
 		log.Error("Error reading body", zap.Error(err))
-		h.HandleHTTPError(ctx, &influxdb.Error{
-			Code: influxdb.EInternal,
-			Op:   "http/handleWrite",
-			Msg:  fmt.Sprintf("unable to read data: %v", err),
-			Err:  err,
-		}, w)
+		handleError(err, influxdb.EInternal, "unable to read data")
+		return
+	}
+
+	// close the reader now that all bytes have been consumed
+	// this will return non-nil in the case of a configured limit
+	// being exceeded
+	if err := in.Close(); err != nil {
+		log.Error("Error reading body", zap.Error(err))
+
+		code := influxdb.EInternal
+		if errors.Is(err, ErrMaxBatchSizeExceeded) {
+			code = influxdb.ETooLarge
```

Comment:
Great to see the use of proper HTTP status codes 🎉

Reply:
You can tell what wikipedia page I spent some time on for this PR 😂